src/context/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { createContext, ReactNode, useContext, FC, useState } from 'react'
+import { useRouter } from 'next/router';
+
+interface Props {
+    children?: ReactNode;
+}
+
+export type ContextType = {
+    state: { toggleEditButton: boolean };
+    actions: { handleToggleButton: () => void }
+}
+const AppContext = createContext<ContextType | null>(null)
```

Comment:
This is quite a generic term `AppContext`
It would be very difficult to differentiate its purpose in the future when we will have multiple contexts.
Could It be more specific

Reply:
What is the purpose of null here?

Reply:
@ankushdharkar what do you say about this comment?